.buildkite/steps/publish.sh

```diff
@@ -43,12 +43,28 @@ else
   echo "Skipping publishing latest, '$BUILDKITE_TAG' doesn't match '$(git describe origin/master --tags --match='v*')'"
 fi
 
-# Publish the most recent commit from each branch
-s3_upload_templates "${BUILDKITE_BRANCH}/"
+publish_for_branch() {
+  local branch="$1"
 
-# Publish each build to a unique URL, to let people roll back to old versions
-s3_upload_templates "${BUILDKITE_BRANCH}/${BUILDKITE_COMMIT}."
+  # Publish the most recent commit from each branch
+  s3_upload_templates "${branch}/"
 
-cat << EOF | buildkite-agent annotate --style "info"
-Published template <a href="https://s3.amazonaws.com/${BUILDKITE_AWS_STACK_TEMPLATE_BUCKET}/${BUILDKITE_BRANCH}/aws-stack.yml">${BUILDKITE_BRANCH}/aws-stack.yml</a>
+  # Publish each build to a unique URL, to let people roll back to old versions
+  s3_upload_templates "${branch}/${BUILDKITE_COMMIT}."
+
+  cat << EOF | buildkite-agent annotate --style "info"
+Published template <a href="https://s3.amazonaws.com/${BUILDKITE_AWS_STACK_TEMPLATE_BUCKET}/${branch}/aws-stack.yml">${branch}/aws-stack.yml</a>
 EOF
+}
+
+if [[ "$BUILDKITE_BRANCH" != "$BUILDKITE_PIPELINE_DEFAULT_BRANCH" ]]; then
+  publish_for_branch "$BUILDKITE_BRANCH"
+
+  exit 0
+fi
+
+# TODO: remove "master" from this list
```

Comment:
This is largely a hypothetical, but if we all forget what this was about, when would the right time to remove "master" be? 6 months from now?

Reply:
i'm a big fan of time bomb comments for this - eg `# If you're reading this and the date is after YYYY-MM-DD the you should delete master from this list`, because it's a concrete call to action, and avoids both the bystander effect and the murky who/when/how of a TODO

Reply:
i remember ages ago finding a rust crate that was able to parse these and automatically fail CI if it found any hanging time bombs... lost to the sands of the internet now though

Reply:
I did make a ticket about it: https://linear.app/buildkite/issue/PDP-1110/remove-publishing-elastic-stack-to-master

I put it in the backlog. I think moving to PDP triage might make it more visible.